joern-cli/frontends/kotlin2cpg/src/test/scala/io/joern/kotlin2cpg/querying/ObjectExpressionTests.scala

```diff
@@ -156,12 +156,12 @@ class ObjectExpressionTests extends KotlinCode2CpgFixture(withOssDataflow = fals
 
     "should contain a correctly lowered representation" in {
       val List(last: Return) =
-        cpg.method.nameExact("withFailListener").block.astChildren.l
```

Comment:
In all these cases I think its better to filter explictly.
E.g., `.astChildren.isReturn.l`. With that, the `@unchecked` annotation becomes unnecessary.

Reply:
good point, i was just blindly following the compiler recommendation to use the automatic rewrite. will change

Reply:
this one from a few lines down is better off with the `@unchecked` annotation though, right?
```
      val List(objExpr: TypeDecl, l: Local, alloc: Call, init: Call, i: Identifier) =
        c.astChildren.isBlock.astChildren.l: @unchecked
```

Reply:
(which actually applies to half the changed lines in this PR)

Reply:
In other tests we explicitly filter with the `.isFoo` API and in case this would result in multiple objects with `.lineNumber` additionaly.

Reply:
That makes sure you actually test the elements meant to be tested. 
(the test style here in this few places is not optimal in first place)

Reply:
Hmm, in the above example wouldn't that result in rather long-winded code? Not sure if that's really worth it, given that this is in test code, and at (test) runtime it'd result in a MatchError if the types don't match. 

For the single-typed lists I'm completely with you.